piccolo_api/crud/endpoints.py

```diff
@@ -612,18 +621,24 @@ async def post_single(
             return Response("Unable to save the resource.", status_code=500)
 
     @apply_validators
-    async def delete_all(
+    async def delete_bulk(
         self, request: Request, params: t.Optional[t.Dict[str, t.Any]] = None
     ) -> Response:
         """
-        Deletes all rows - query parameters are used for filtering.
+        Bulk deletes rows - query parameters are used for filtering.
         """
         params = self._clean_data(params) if params else {}
         split_params = self._split_params(params)
+        split_params_ids = split_params.ids.split(",")
 
         try:
             query = self._apply_filters(
-                self.table.delete(force=True), split_params
+                self.table.delete().where(
+                    self.table._meta.primary_key.is_in(
+                        [int(item) for item in split_params_ids]
+                    )
+                ),
```

Comment:
Here we should check if there are any valid ids before applying the filter, otherwise it will fail.

```python
query = self.table.delete()
ids = [int(item) for item in split_params_ids]
if ids:
    query = query.where(self.table._meta.primary_key.is_in(ids))
query = self._apply_filters(query, split_params)
```

Reply:
Yes, you're right! I changed the `delete_bulk` method to work only with filter params (e.g. name param), only with `__ids` param and to work together for additional safety. 

```python
try:
    query: t.Union[
        Select, Count, Objects, Delete
    ] = self.table.delete()
    try:
        # if __ids query param is not empty
        ids = [int(item) for item in split_params_ids]
        query_ids = query.where(
            self.table._meta.primary_key.is_in(ids)
        )
        query = self._apply_filters(query_ids, split_params)
    except ValueError:
        # if __ids query param is empty
        query = self._apply_filters(query, split_params)
  except MalformedQuery as exception:
    return Response(str(exception), status_code=400)
```
It's a little complicated but it works.